opendatahub-io/odh-dashboard
# Added accelerator edit page

closes: https://github.com/opendatahub-io/odh-dashboard/issues/1375

## Description



- created a accelerator routing component to cover all the pages
- broke manage accelerator into a edit component and the core component
- edit fetches the existing profile and handles any errors there
- core manage page is a basic form with
   - name, identifier, description, enabled, and tolerations
   - tolerations is a table
       - adding new/editing tolerations opens a modal with toleration fields. some are drop downs, some are text, key is required, and seconds is a radio that can be set to forever or a number
       - toleration modal does not make network calls
       - toleration modal has non blocking info alerts when invalid combo of inputs is detected
- new k8s api functions for update, get, and post
- new function for generating k8s yaml

**Error**
<img width="1290" alt="Screenshot 2023-11-01 at 2 49 19 PM" src="https://github.com/opendatahub-io/odh-dashboard/assets/12587674/436575f8-6c9d-4b5c-8ae3-70f3af19e732">

**Accelerator Page**
![Screenshot 2023-11-03 at 1 52 39 PM](https://github.com/opendatahub-io/odh-dashboard/assets/12587674/b95f2d41-b306-4a3d-a9df-7b396696d41c)

**Helper texts on accelerator page**
![Screenshot 2023-11-03 at 1 52 49 PM](https://github.com/opendatahub-io/odh-dashboard/assets/12587674/28b44dcb-ca45-40c4-9c04-592b9ca512b6)
<img width="647" alt="image" src="https://github.com/opendatahub-io/odh-dashboard/assets/12587674/7450220a-adfd-4a3e-ba51-c502d63294e4">


**Toleration Modal**
![Screenshot 2023-11-03 at 1 53 03 PM](https://github.com/opendatahub-io/odh-dashboard/assets/12587674/32381855-ffaa-480d-ba5f-93c93f7f11d8)
![Screenshot 2023-11-03 at 1 53 11 PM](https://github.com/opendatahub-io/odh-dashboard/assets/12587674/cc808087-5caf-49ac-8486-9298925ec5ba)
![Screenshot 2023-11-03 at 1 53 18 PM](https://github.com/opendatahub-io/odh-dashboard/assets/12587674/4d8302d0-3761-4cfd-bf3b-91b2a77607bb)
![Screenshot 2023-11-03 at 1 53 30 PM](https://github.com/opendatahub-io/odh-dashboard/assets/12587674/f2582410-c2c1-43c9-83f5-7a53fa4889bd)
![Screenshot 2023-11-03 at 1 54 46 PM](https://github.com/opendatahub-io/odh-dashboard/assets/12587674/8c9beb11-3660-467c-aeaf-c310d3b3ecc0)


## How Has This Been Tested?




- Go to /acceleratorProfiles/manage to open the create page
- play around with the form
   - add and remove tolerations from the table
   - make sure toleration form is reflected in the core form
   - submit when satisfied with form
- edit the accelerator by going to /acceleratorProfiles/manage/{resource-id}
   - edit the name to something else
   - remove tolerations
   - edit other fields
   - submit and edit it again. make sure all is reflected
- edit a profile that does not exist, see error page
- go to routes such as /acceleratorProfiles/wrong or /acceleratorProfiles/manage/id/not/a/route and make sure you are redirected to /acceleratorProfiles

## Test Impact


tests exist for the page in all three states (create, edit, error)

## Request review criteria:



Self checklist (all need to be checked):

If you have UI changes: 


After the PR is posted & before it merges:
